Setting up SSF is not idempotent

## Current Behavior

Multiple things will cause errors like:
 ```
The Secret "signing-secrets" is invalid: data: Forbidden: field is immutable when `immutable` is set
make: *** [Makefile:38: setup-tekton-chains] Error 1
```

on a second run.

## Expected Behavior

Ideally anything we do to install and setup SSF should be idempotent. If it can't be idempotent for some reason then it should be a no-op

## Steps to Reproduce

1. run `make quickstart`
2. run `make quickstart` a second time

--- a/platform/11-tekton-chains.sh
+++ b/platform/11-tekton-chains.sh
@@ -6,7 +6,10 @@ GIT_ROOT=$(git rev-parse --show-toplevel)
 # Setup tekton Chains
 
 # Install Chains.
-kubectl apply --filename "$GIT_ROOT"/platform/vendor/tekton/chains/release.yaml
+SECRETS=$(kubectl get secret signing-secrets -n tekton-chains -o jsonpath='{.immutable}' 2> /dev/null || true)
+if [ "$SECRETS" != "true" ]; then
+      kubectl apply --filename "$GIT_ROOT"/platform/vendor/tekton/chains/release.yaml
+fi
 kubectl rollout status -n tekton-chains deployment/tekton-chains-controller
 
 # Patch chains to generate in-toto provenance and store output in OCI

